### PR TITLE
test: Add test for commit function

### PR DESCRIPTION
**Pull request by lynnagara**
Adds an end to end test for committing offsets in the Kafka consumer. Also adds helper functions for creating and deleting Kafka topics. Needs a running Kafka on `localhost:9092` to work.

Now we run Kafka and Zookeeper in CI. Since docker is not available in macos, removed that test suite.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -78,6 +78,25 @@ jobs:
...
+            --name sentry_kafka \
+            -d --network host \
+            -e KAFKA_ZOOKEEPER_CONNECT=127.0.0.1:2181 \
+            -e KAFKA_LISTENERS=INTERNAL://0.0.0.0:9093,EXTERNAL://0.0.0.0:9092 \
+            -e KAFKA_ADVERTISED_LISTENERS=INTERNAL://127.0.0.1:9093,EXTERNAL://127.0.0.1:9092 \
+            -e KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT \
+            -e KAFKA_INTER_BROKER_LISTENER_NAME=INTERNAL \
+            -e KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1 \
+            confluentinc/cp-kafka:6.2.0
+
```
Comment (volokluev) [inline .github/workflows/ci.yml:99]: Can we have this in a docker-compose or just anything that makes it easy to run it locally without copy-pasting it into your command line?

**Review by volokluev — APPROVED** [review]
approved but I don't have a good understanding of how to run this locally right now

File: src/backends/kafka/mod.rs [hunk 2]
```diff
@@ -297,8 +331,42 @@ mod tests {
         consumer.subscribe(&[topic], my_callbacks).unwrap();
     }
 
-    #[test]
-    fn test_commit() {}
+    #[tokio::test]
+    async fn test_commit() {
+        create_topic("test", 1).await;
```
Comment (volokluev) [inline src/backends/kafka/mod.rs:336]: what happens if the topic already exists?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lynnagara) [reply .github/workflows/ci.yml:99]: Currently the best way to run it locally for Sentry developers is with Sentry devservices. It should just work. This is the same as what Snuba + Python Arroyo does. At this point, I would avoid adding something separate for those not running devservices.

**Review by volokluev — APPROVED** [review]

**Pull request merged**